presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java

```diff
@@ -751,4 +766,78 @@ public long getRetainedSizeInBytes()
             return INSTANCE_SIZE + sizeOf(outputPositions);
         }
     }
+
+    private static final class MissingFieldStreamReader
+            implements SelectiveStreamReader
+    {
+        private static final int INSTANCE_SIZE = ClassLayout.parseClass(MissingFieldStreamReader.class).instanceSize();
+        private final Type outputType;
+        private int[] outputPositions;
+        private int outputPositionCount;
+
+        MissingFieldStreamReader(Optional<Type> type)
+        {
+            this.outputType = type.isPresent() ? type.get() : null;
+        }
+
+        @Override
+        public int read(int offset, int[] positions, int positionCount)
+        {
+            outputPositions = initializeOutputPositions(outputPositions, positions, positionCount);
+            outputPositionCount = positionCount;
+            return outputPositionCount;
+        }
+
+        @Override
+        public int[] getReadPositions()
```

Comment:
Is this every used? Can it be implemented to throw NotImplementedException? Then, outputPositions variable can be removed and getRetainedSizeInBytes can become empty.

Reply:
This method is called for all the fields, so we will need it